tests/Aql/InsertExecutorTest.cpp

```diff
@@ -0,0 +1,428 @@
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Markus Pfeiffer
+////////////////////////////////////////////////////////////////////////////////
+
+#include "../Mocks/Servers.h"
+#include "QueryHelper.h"
+#include "gtest/gtest.h"
+
+#include "Aql/Query.h"
+#include "Aql/VelocyPackHelper.h"
+#include "RestServer/QueryRegistryFeature.h"
+
+#include <velocypack/Buffer.h>
+#include <velocypack/Builder.h>
+#include <velocypack/Iterator.h>
+#include <velocypack/Slice.h>
+#include <velocypack/velocypack-aliases.h>
+
+#include "../IResearch/IResearchQueryCommon.h"
+
+#include "Logger/LogMacros.h"
+
+using namespace arangodb;
+using namespace arangodb::aql;
+
+namespace arangodb {
+namespace tests {
+namespace aql {
+
+class InsertExecutorTest : public ::testing::Test {
+ protected:
+  mocks::MockAqlServer server;
+  TRI_vocbase_t& vocbase;
+
+  std::string const collectionName = "UnitTestCollection";
+  std::string const checkQuery =
+      "FOR i IN " + collectionName + " SORT i.value RETURN i.value";
+
+ public:
+  InsertExecutorTest() : vocbase(server.getSystemDatabase()) {}
+
+  void SetUp() override {
+    SCOPED_TRACE("SetUp");
+    auto info = VPackParser::fromJson("{\"name\": \"" + collectionName + "\"}");
+    auto collection = vocbase.createCollection(info->slice());
+    ASSERT_NE(collection.get(), nullptr) << "Failed to create collection";
+  }
+};
+
+class InsertExecutorTestCount : public InsertExecutorTest,
+                                public ::testing::WithParamInterface<size_t> {
+ public:
+  size_t nDocs;
+  std::string nDocsString;
+
+ public:
+  InsertExecutorTestCount()
+      : nDocs(GetParam()), nDocsString(std::to_string(GetParam())) {}
+};
+
+class InsertExecutorTestCounts
+    : public InsertExecutorTest,
+      public ::testing::WithParamInterface<std::vector<size_t>> {};
+
+TEST_F(InsertExecutorTest, basic) {
+  std::string query = R"(INSERT { value: 1 } IN )" + collectionName;
+  AssertQueryHasResult(vocbase, query, VPackSlice::emptyArraySlice());
+
+  auto expected = VPackParser::fromJson(R"([1])");
+  AssertQueryHasResult(vocbase, checkQuery, expected->slice());
+}
+
+TEST_F(InsertExecutorTest, insert_but_not_rev) {
+  std::string query = R"(INSERT { _key: "foo", _rev: "bar" } IN )" + collectionName;
+  AssertQueryHasResult(vocbase, query, VPackSlice::emptyArraySlice());
+
+  auto const bindParameters = VPackParser::fromJson("{ }");
+
+  auto queryResult =
+      arangodb::tests::executeQuery(vocbase,
+                                    "FOR d IN " + collectionName + " RETURN d",
+                                    bindParameters);
+  ASSERT_TRUE(queryResult.ok());
+  auto slice = queryResult.data->slice();
+  ASSERT_TRUE(slice.isArray());
+  ASSERT_EQ(slice.length(), 1);
+
+  auto doc = slice.at(0);
+  ASSERT_NE(arangodb::basics::VelocyPackHelper::getStringValue(doc, "_rev", ""),
```

Comment:
I think this test is not correct.
It tries to write "bar", and asserts that it has not written "foo", where I assume it wanted to make sure "bar" is not written by server generated.